src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -1,15 +1,30 @@
 package de.bund.digitalservice.useid.widget
 
+import de.bund.digitalservice.useid.config.ContentSecurityPolicyProperties
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.http.server.reactive.ServerHttpResponse
 import org.springframework.stereotype.Controller
-import org.springframework.ui.Model
 import org.springframework.web.bind.annotation.GetMapping
 import reactor.core.publisher.Mono
 
 @Controller
-class WidgetController {
+class WidgetController(
+    private val contentSecurityPolicyProperties: ContentSecurityPolicyProperties
+) {
     @GetMapping("/widget")
-    fun widget(model: Model): Mono<String> = Mono.just("widget")
+    fun widget(serverHttpResponse: ServerHttpResponse, serverHttpRequest: ServerHttpRequest): Mono<String> {
+        val host = serverHttpRequest.uri.host
+        val allowedHost = contentSecurityPolicyProperties.domainIsAllowed(host)
+
+        if (allowedHost) {
```

Comment:
What happens in the else case?

Reply:
I updated the logic, it nows filtered in `src/main/kotlin/de/bund/digitalservice/useid/widget/ContentSecurityPolicyFilter.kt` and when else, its response only returns `self` which only our domain can load the widget